### PR TITLE
Implement `conway governance committee create-cold-key-resignation` certificate

### PR DESCRIPTION
**Pull request by newhoggy**
# Changelog

```yaml
- description: |
   Implement `conway governance committee create-cold-key-resignation` certificate
# uncomment types applicable to the change:
  type:
  # - breaking       # the API has changed in a breaking way
  - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - documentation  # change in code docs, haddocks...
```

# Context

Implements https://github.com/input-output-hk/cardano-cli/issues/4

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - round trip tests
  - integration tests
  See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] The changelog section in the PR is updated to describe the change
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/test/cardano-cli-golden/files/golden/help/experimental_governance_committee_create-cold-key-resignation-certificate.cli [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+                                                                                               | --cc-cold-key-file FILE
+                                                                                               | --cc-cold-key-hash STRING
+                                                                                               )
+                                                                                               --out-file FILE
+
+  Create cold key resignation certificate for a Constitutional Committee Member
+
+Available options:
+  --cc-cold-key STRING     Constitutional Committee cold key (hex-encoded).
+  --cc-cold-key-file FILE  Filepath of the Consitutional Committee cold key.
```
Comment (Jimbo4350) [inline cardano-cli/test/cardano-cli-golden/files/golden/help/experimental_governance_committee_create-cold-key-resignation-certificate.cli]: We should stick to the current cli format: `--cold-verification-key-file`. The `cc` is redundant.

**Review by Jimbo4350 — DISMISSED** [review]

File: cardano-cli/test/cardano-cli-golden/files/golden/help/experimental_governance_committee_create-cold-key-resignation-certificate.cli (hunk 1 above)
Comment (newhoggy) [reply cardano-cli/test/cardano-cli-golden/files/golden/help/experimental_governance_committee_create-cold-key-resignation-certificate.cli]: Done!

File: cardano-cli/src/Cardano/CLI/EraBased/Legacy.hs [hunk 2]
```diff
@@ -0,0 +1,3341 @@
...
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Cardano.CLI.EraBased.Legacy
+  ( -- * CLI command parser
+    parseLegacyCommands
+
+    -- * CLI command and flag types
+  , module Cardano.CLI.Commands.Legacy
```
Comment (carbolymer) [inline cardano-cli/src/Cardano/CLI/EraBased/Legacy.hs]: What's the difference between this and erabased legacy?

File: cardano-cli/src/Cardano/CLI/EraBased/Legacy.hs
Comment (carbolymer) [inline cardano-cli/src/Cardano/CLI/EraBased/Legacy.hs:1]: These are all parsers, shouldn't they go into `EraBased.Options` module?

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Governance/Committee.hs [hunk 3]
```diff
@@ -88,3 +89,27 @@ pGovernanceCommitteeKeyHash era = do
     pCmd w =
       GovernanceCommitteeKeyHash w
         <$> pVerificationKeyFileIn
+
+pGovernanceCommitteeCreateColdKeyResignationCertificate :: ()
+  => CardanoEra era
+  -> Maybe (Parser (GovernanceCommitteeCmds era))
+pGovernanceCommitteeCreateColdKeyResignationCertificate =
+  featureInEra
```
Comment (carbolymer) [inline cardano-cli/src/Cardano/CLI/EraBased/Options/Governance/Committee.hs]: why not `maybeFeatureInEra` instead?

**Review by carbolymer — APPROVED** [review]

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Governance/Committee.hs (hunk 3 above)
Comment (newhoggy) [reply cardano-cli/src/Cardano/CLI/EraBased/Options/Governance/Committee.hs]: `maybeFeatureInEra` isn't the right function to use.  There is a theoretically a better function to use than `featureInEra`, but it hasn't been written unfortunately.

File: cardano-cli/src/Cardano/CLI/EraBased/Legacy.hs
Comment (newhoggy) [reply cardano-cli/src/Cardano/CLI/EraBased/Legacy.hs:1]: You are right.  Let me make a new PR for this.

File: cardano-cli/src/Cardano/CLI/EraBased/Legacy.hs (hunk 2 above)
Comment (newhoggy) [reply cardano-cli/src/Cardano/CLI/EraBased/Legacy.hs]: I think that's a stray commit that found its way into the PR.  I don't see it anymore.

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Governance/Committee.hs (hunk 3 above)
Comment (carbolymer) [reply cardano-cli/src/Cardano/CLI/EraBased/Options/Governance/Committee.hs]: why not?

Comment (newhoggy) [reply cardano-cli/src/Cardano/CLI/EraBased/Options/Governance/Committee.hs]: Using `maybeFeatureInEra` as suggested.

**Pull request merged**